src/components/playlisteditor/playlisteditor.ts

```diff
@@ -0,0 +1,319 @@
+import { BaseItemKind } from '@jellyfin/sdk/lib/generated-client/models/base-item-kind';
+import { ItemSortBy } from '@jellyfin/sdk/lib/generated-client/models/item-sort-by';
+import { getItemsApi } from '@jellyfin/sdk/lib/utils/api/items-api';
+import { getPlaylistsApi } from '@jellyfin/sdk/lib/utils/api/playlists-api';
+import escapeHtml from 'escape-html';
+
+import dom from 'scripts/dom';
+import globalize from 'scripts/globalize';
+import { currentSettings as userSettings } from 'scripts/settings/userSettings';
+import { PluginType } from 'types/plugin';
+import { toApi } from 'utils/jellyfin-apiclient/compat';
+
+import dialogHelper from '../dialogHelper/dialogHelper';
+import loading from '../loading/loading';
+import layoutManager from '../layoutManager';
+import { playbackManager } from '../playback/playbackmanager';
+import { pluginManager } from '../pluginManager';
+import { appRouter } from '../router/appRouter';
+import ServerConnections from '../ServerConnections';
+
+import 'elements/emby-button/emby-button';
+import 'elements/emby-input/emby-input';
+import 'elements/emby-button/paper-icon-button-light';
+import 'elements/emby-select/emby-select';
+
+import 'material-design-icons-iconfont';
+import '../formdialog.scss';
+
+interface DialogElement extends HTMLDivElement {
+    submitted?: boolean
+}
+
+interface PlaylistEditorOptions {
+    items: string[],
+    serverId: string,
+    enableAddToPlayQueue?: boolean,
+    defaultValue?: string
+}
+
+let currentServerId: string;
+
+function onSubmit(this: HTMLElement, e: Event) {
+    const panel = dom.parentWithClass(this, 'dialog') as DialogElement | null;
+
+    if (panel) {
+        const playlistId = panel.querySelector<HTMLSelectElement>('#selectPlaylistToAddTo')?.value;
+
+        loading.show();
+
+        if (playlistId) {
+            userSettings.set('playlisteditor-lastplaylistid', playlistId);
+            addToPlaylist(panel, playlistId)
+                ?.catch(err => {
```

Comment:
Doesn't a "failed" optional chain make `finally` unreachable?
I think if we guarantee Promise will be returned, we can drop the optional chaining.

Reply:
Good call. Should be fixed now.